src/gui/src/gui.cpp

```diff
@@ -856,6 +856,11 @@ void Gui::addFocusNet(odb::dbNet* net)
   main_window->getLayoutViewer()->addFocusNet(net);
 }
 
+void Gui::addRouteGuides(odb::dbNet* net)
+{
+  main_window->getLayoutViewer()->addRouteGuides(net);
+}
```

Comment:
DRT or GRT should probably have these. and Only register the renderer when something is added and remove it when it's empty.